server/src/main/java/io/crate/replication/logical/action/TransportDropSubscriptionAction.java

```diff
@@ -21,130 +21,283 @@
-            ),
-            new AckedClusterStateUpdateTaskChain.ChainableTask(
-                removeSubscriptionSetting,
-                "Please run command DROP SUBSCRIPTION again",
-                "remove-subscription-setting"
-            ),
-            new AckedClusterStateUpdateTaskChain.ChainableTask(
-                openSubscriptionIndices,
-                "Please run command ALTER TABLE OPEN for subscribed tables",
-                "open-subscription-indices"
-            )
-        );
 
-        AckedClusterStateUpdateTaskChain ackedClusterStateUpdateTaskChain =
-            new AckedClusterStateUpdateTaskChain(tasks, clusterService, listener, request, Priority.HIGH);
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
 
-        ackedClusterStateUpdateTaskChain.startTaskChain();
+    @Override
+    protected WriteUserResponse read(StreamInput in) throws IOException {
+        return new WriteUserResponse(in);
     }
 
     @Override
-    public ClusterStateTaskExecutor<DropSubscriptionRequest> clusterStateTaskExecutor(DropSubscriptionRequest request) {
-        return new DDLClusterStateTaskExecutor<>() {
-            @Override
-            protected ClusterState execute(ClusterState currentState, DropSubscriptionRequest dropSubscriptionRequest) throws Exception {
-                Metadata currentMetadata = currentState.metadata();
-                Metadata.Builder mdBuilder = Metadata.builder(currentMetadata);
-
-                SubscriptionsMetadata oldMetadata = (SubscriptionsMetadata) mdBuilder.getCustom(SubscriptionsMetadata.TYPE);
-                if (oldMetadata != null && oldMetadata.subscription().containsKey(request.name())) {
-
-                    // LogicalReplicationService will react to cluster change events and will stop the replication.
-                    SubscriptionsMetadata newMetadata = SubscriptionsMetadata.newInstance(oldMetadata);
-                    newMetadata.subscription().remove(request.name());
-                    assert !newMetadata.equals(oldMetadata) : "must not be equal to guarantee the cluster change action";
-                    mdBuilder.putCustom(SubscriptionsMetadata.TYPE, newMetadata);
-
-                    return ClusterState.builder(currentState).metadata(mdBuilder).build();
-                } else if (request.ifExists() == false) {
-                    throw new SubscriptionUnknownException(request.name());
-                }
+    protected void masterOperation(DropSubscriptionRequest request, ClusterState state, ActionListener<AcknowledgedResponse> listener) throws Exception {
+        List<IndexMetadata> subscriptionIndices = getSubscriptionIndices(request.name(), state.metadata().indices());
+        List<OpenCloseTable> openCloseTables = getSubscriptionTables(request.name(), schemas);
 
-                return currentState;
-            }
-        };
+        submitCloseSubscriptionsTablesTask(openCloseTables)
```

Comment:
initially I had check from the second step here
 `if (oldMetadata != null && oldMetadata.subscription().containsKey(subscriptionName)) {`
but actually close will not do anything (will get empty list) so should be fine.